store/src/main/scala/geotrellis/store/cog/COGCollectionLayerReader.scala

```diff
@@ -234,7 +234,7 @@ object COGCollectionLayerReader {
           .flatten
           .flatMap { case (spatialKey, overviewIndex, _, seq) =>
             val key = baseKey.setComponent(spatialKey)
-            val tiff = GeoTiffReader[V].read(uri, streaming = true).getOverview(overviewIndex)
+            val tiff = GeoTiffReader[V].read(rangeReader, streaming = true).getOverview(overviewIndex)
```

Comment:
Hm I didnt have a closer look to these changes but I am not sure that we can reuse the same range reader for different reads O:

Reply:
Hmm,  there were a number of test failures in the COG reading/writing functionality on one of the test runs. I'll dig deeper a bit later, but in the meantime I pushed a new commit that reverts to using a new RangeReader instance for each read in order to let Travis get working on the test suite.

Reply:
Fun. Upon further digging, switching to RangeReader might be causing some backwards incompatibility. According to the [FileRangeReader](https://github.com/locationtech/geotrellis/blob/master/util/src/main/scala/geotrellis/util/FileRangeReaderProvider.scala#L25-L48), I'd expect to be able to construct FileRangeReaders by passing file URIs with the `file://`, `file:` and empty prefixes. If I attempt that in a console, only the empty prefix works:
```

scala> val rr = RangeReader("/Users/andrew/src/geotrellis/raster/data/geotiff-test-files/cloud_images/blue/1.TIF")
rr: geotrellis.util.RangeReader = geotrellis.util.FileRangeReader@2d2daf13

scala> val rr = RangeReader("file:/Users/andrew/src/geotrellis/raster/data/geotiff-test-files/cloud_images/blue/1.TIF")
scala.MatchError: sun.net.www.protocol.file.FileURLConnection:file:/Users/andrew/src/geotrellis/raster/data/geotiff-test-files/cloud_images/blue/1.TIF (of class sun.net.www.protocol.file.FileURLConnection)
  at scalaj.http.HttpRequest.scalaj$http$HttpRequest$$doConnection(Http.scala:351)
  at scalaj.http.HttpRequest.exec(Http.scala:343)
  at scalaj.http.HttpRequest.asString(Http.scala:492)
  at geotrellis.spark.store.http.util.HttpRangeReader.<init>(HttpRangeReader.scala:42)
  at geotrellis.spark.store.http.util.HttpRangeReader$.apply(HttpRangeReader.scala:105)
  at geotrellis.spark.store.http.util.HttpRangeReader$.apply(HttpRangeReader.scala:97)
  at geotrellis.spark.store.http.util.HttpRangeReaderProvider.rangeReader(HttpRangeReaderProvider.scala:34)
  at geotrellis.spark.store.http.util.HttpRangeReaderProvider.rangeReader(HttpRangeReaderProvider.scala:24)
  at geotrellis.util.RangeReader$.apply(RangeReader.scala:63)
  at geotrellis.util.RangeReader$.apply(RangeReader.scala:65)
  ... 40 elided

scala> val rr = RangeReader("file:///Users/andrew/src/geotrellis/raster/data/geotiff-test-files/cloud_images/blue/1.TIF")
scala.MatchError: sun.net.www.protocol.file.FileURLConnection:file:/Users/andrew/src/geotrellis/raster/data/geotiff-test-files/cloud_images/blue/1.TIF (of class sun.net.www.protocol.file.FileURLConnection)
  at scalaj.http.HttpRequest.scalaj$http$HttpRequest$$doConnection(Http.scala:351)
  at scalaj.http.HttpRequest.exec(Http.scala:343)
  at scalaj.http.HttpRequest.asString(Http.scala:492)
  at geotrellis.spark.store.http.util.HttpRangeReader.<init>(HttpRangeReader.scala:42)
  at geotrellis.spark.store.http.util.HttpRangeReader$.apply(HttpRangeReader.scala:105)
  at geotrellis.spark.store.http.util.HttpRangeReader$.apply(HttpRangeReader.scala:97)
  at geotrellis.spark.store.http.util.HttpRangeReaderProvider.rangeReader(HttpRangeReaderProvider.scala:34)
  at geotrellis.spark.store.http.util.HttpRangeReaderProvider.rangeReader(HttpRangeReaderProvider.scala:24)
  at geotrellis.util.RangeReader$.apply(RangeReader.scala:63)
  at geotrellis.util.RangeReader$.apply(RangeReader.scala:65)
  ... 40 elided
```

If I create `URI` instances of all three of the above URIs, I correctly get `uri.getScheme` equal to `file` for the first two and `null` for the last, which is also correct according to the logic in `FileRangeReader.canProcess`.

Reply:
@CloudNiner hmmm how is that related to https://github.com/locationtech/geotrellis/issues/3031 ?

Reply:
After further review, yup, this appears to be a bug. The HttpRangeReaderProvider.canProcess method is allowing through URLs with non-http[s] schemes. Pushed a tentative fix in https://github.com/locationtech/geotrellis/pull/3093/commits/27a87e952a2b29f979606154384aa88adb6ae527

Reply:
@pomadchin success. Ready for another look.